apps/native-component-list/src/screens/Image/ImageGeneratePlaceholders.tsx

```diff
@@ -0,0 +1,217 @@
+import { Image } from 'expo-image';
+import { useState, useCallback } from 'react';
+import { StyleSheet, View, ScrollView, Text, Button, ActivityIndicator } from 'react-native';
+
+const uri = 'https://picsum.photos/seed/175/300/200';
+
+enum PlaceholderType {
+  Blurhash,
+  Thumbhash,
+}
+enum SourceType {
+  Uri,
+  SharedRef,
+}
+
+interface PlaceholderState {
+  type: PlaceholderType;
+  source: SourceType;
+  value: string | null;
+  loading: boolean;
+  error: string | null;
+}
+
+export default function ImageGeneratePlaceholders() {
+  const initialStates: PlaceholderState[] = [
+    {
+      type: PlaceholderType.Blurhash,
+      source: SourceType.Uri,
+      value: null,
+      loading: false,
+      error: null,
+    },
+    {
+      type: PlaceholderType.Blurhash,
+      source: SourceType.SharedRef,
+      value: null,
+      loading: false,
+      error: null,
+    },
+    {
+      type: PlaceholderType.Thumbhash,
+      source: SourceType.Uri,
+      value: null,
+      loading: false,
+      error: null,
+    },
+    {
+      type: PlaceholderType.Thumbhash,
+      source: SourceType.SharedRef,
+      value: null,
+      loading: false,
+      error: null,
+    },
+  ];
```

Comment:
```suggestion
const initialStates: PlaceholderState[] = [
  {
    type: PlaceholderType.Blurhash,
    source: SourceType.Uri,
    value: null,
    loading: false,
    error: null,
  },
  {
    type: PlaceholderType.Blurhash,
    source: SourceType.SharedRef,
    value: null,
    loading: false,
    error: null,
  }, 
  {
    type: PlaceholderType.Thumbhash,
    source: SourceType.Uri,
    value: null,
    loading: false,
    error: null,
  },
  {
    type: PlaceholderType.Thumbhash,
    source: SourceType.SharedRef,
    value: null,
    loading: false,
    error: null,
  },
];

export default function ImageGeneratePlaceholders() {  
```
Let's move the default state to the top level.